doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,273 @@
+.. _can_interface:
+
+Controller Area Network (CAN)
+#############################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+
+Controller Area Network is a two-wire serial bus specified by the
+Bosch CAN Specification, Bosch CAN with Flexible Data-Rate specification and the
+ISO 11898-1:2003 standard.
+CAN is mostly known for its application in the automotive domain. However, it
+is also used in home and industrial automation and other products.
+
+A so-called CAN transceiver connects the two bus lines, CAN High (CAN H) and
+CAN Low (CAN L), from the CAN network to the CAN controller.
+The send wire from the controller to the transceiver is called CAN TX, and the
+receive wire is called CAN RX. These wires use the logic levels whereas the bus
+level is interpreted differential between CAN H and CAN L.
+Both lines, CAN H and CAN L, at the same level, is called the recessive state.
+This state is seen as a logical one, which is also the idle state.
+The other state is called the dominant state and is interpreted as a logical
+zero value. To write a dominant bit to the bus, open-drain transistors tight
+CAN H to Vdd and CAN L to ground. The first and last node use a 120 ohm resistor
+between CAN H and CAN L to terminate the bus.
+The termination avoids reflections on the bus and keeps CAN L and CAN H at the
+same level in idle state.
+The dominant state always overrides the recessive state.
+This structure is called a wired-AND.
```

Comment:
I wonder if a small picture would help to understand the thing here. The explanation above sounds complicated if one is not a CAN expert.

Reply:
I have pictures from my masters' thesis. Not sure if I can use them here. I would somehow need to add a copyright to not run into problems regarding plagiarism when submitting the thesis.

Reply:
I could add them in a few months when the thesis was accepted :-)

Reply:
Used my LaTex tikz skills to drow some nice pictures :-)

Reply:
I noticed the .svg files, and started to wonder that if you did the pictures using LaTex, would it be possible to have the originals also here? That way if the images need changes, it is possible to re-use the original files instead of starting from scratch.

Reply:
If you haven't found the draw.io web site, that's a dandy tool for editing and saving SVG images in a form that can be edited (again) using draw.io, without creating another file.

Reply:
I think we should not add the LaTex sources. These are six files that are not compiled into the doc.